py/tools/venv_bin/src/main.rs

```diff
@@ -115,76 +115,65 @@ struct VenvArgs {
 
 fn venv_cmd_handler(args: VenvArgs) -> miette::Result<()> {
     let pth_file = py::PthFile::new(&args.pth_file, args.pth_entry_prefix);
-    match args.mode {
+    if let VenvMode::DynamicSymlink = args.mode {
```

Comment:
By switching from a match with one arm acting and the other arm doing-work-then-matching-again to `if let` with a `return`, I took out one level of indentation with no other code changes. The second match still needs the `unreachable!` in it because rustc doesn't have the ability to statically strip enum variants in cases like these. Annoying but c'est la vie. Beats writing Haskal